### PR TITLE
Fixing slowness in saved post fetching. #4756

### PR DESCRIPTION
**Pull request by dessalines**
Based on the weird slowness below, and the extra complication from all these `BoxableExpressions`, I think I'm going to recommend that we go back to the simplicity of using left joins, rather than subqueries.

Check out my comments below tho.

cc @dullbananas

### CODE REVIEW

File: crates/db_views/src/post_view.rs [hunk 1]
```diff
@@ -100,17 +99,6 @@ fn queries<'a>() -> Queries<
...
 
-  let is_saved = |person_id| {
-    post_saved::table
-      .filter(
-        post_aggregates::post_id
-          .eq(post_saved::post_id)
-          .and(post_saved::person_id.eq(person_id)),
-      )
-      .select(post_saved::published.nullable())
-      .single_value()
```
Comment (dessalines) [inline crates/db_views/src/post_view.rs:111]: I tried various things, but this was the culprit.

When I changed this subquery from a select to `exists`, like the `is_read` function just below it, the query is instant. But as soon as a select is added, it slows down.

File: crates/db_views/src/post_view.rs [hunk 2]
```diff
@@ -237,6 +217,13 @@ fn queries<'a>() -> Queries<
       .inner_join(person::table)
       .inner_join(community::table)
       .inner_join(post::table)
+      .left_join(
+        post_saved::table.on(
+          post_aggregates::post_id
+            .eq(post_saved::post_id)
+            .and(post_saved::person_id.eq(my_person_id.unwrap_or(PersonId(-1)))),
```
Comment (dessalines) [inline crates/db_views/src/post_view.rs:224]: Going back to a left_join, from the subquery, instantly sped it up, and now it has the same speed as `saved_only = false`

**Review by dessalines — COMMENTED** [review]
Based on this weird subquery slowness, I think it'd be a good idea for me to convert all of these subqueries in `post_view` and `comment_view` back to left joins. It would also eliminate a lot of the complicated logic around these boxableexpressions also.

But I'd like to get @dullbananas thoughts before I continue with that.

**Comment by dullbananas** [comment]
I'm already changing everything to left joins in #4459, mainly because it's required for selecting multiple columns. For post view, it will look like this:

```
.left_join(actions(
  community_actions::table,
  my_person_id,
  post_aggregates::community_id,
))
.left_join(actions(
  person_actions::table,
  my_person_id,
  post_aggregates::creator_id,
))
.left_join(actions(
  post_actions::table,
  my_person_id,
  post_aggregates::post_id,
))
.left_join(actions(
  instance_actions::table,
  my_person_id,
  post_aggregates::instance_id,
))
.left_join(actions_alias(
  creator_community_actions,
  post_aggregates::creator_id,
  post_aggregates::community_id,
))
```

**Review by dullbananas — APPROVED** [review]

**Comment by dessalines** [comment]
Thx. In that case, I'll leave my changes minimal since your PR will eventually overwrite mine anyway.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**